pkg/orchestrator/pipelines_test.go

```diff
@@ -302,5 +313,140 @@ func TestPipelineOrchestrator_Delete_PipelineDoesntExist(t *testing.T) {
 		Return(nil, wantErr)
 
 	err := orc.Pipelines.Delete(ctx, uuid.NewString())
-	assert.Error(t, err)
+	is.True(err != nil)
+}
+
+func TestPipelineOrchestrator_UpdateDLQ_Success(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+	db := &inmemory.DB{}
+	ctrl := gomock.NewController(t)
+	plsMock, consMock, procsMock, pluginMock := newMockServices(t)
+
+	pluginDispenser := pmock.NewDispenser(ctrl)
+
+	plBefore := &pipeline.Instance{
+		ID:     uuid.NewString(),
+		Status: pipeline.StatusSystemStopped,
+		Config: pipeline.Config{Name: "test-pipeline"},
+		DLQ: pipeline.DLQ{
+			Plugin:              "old-plugin",
+			Settings:            map[string]string{"foo": "bar"},
+			WindowSize:          4,
+			WindowNackThreshold: 1,
+		},
+	}
+	newDLQ := pipeline.DLQ{
+		Plugin:              "new-plugin",
+		Settings:            map[string]string{"baz": "qux"},
+		WindowSize:          2,
+		WindowNackThreshold: 0,
+	}
+	want := &pipeline.Instance{
+		ID:     plBefore.ID,
+		Status: pipeline.StatusSystemStopped,
+		Config: pipeline.Config{Name: "test-pipeline"},
+		DLQ:    newDLQ,
+	}
+
+	orc := NewOrchestrator(db, log.Nop(), plsMock, consMock, procsMock, pluginMock)
+	plsMock.EXPECT().
+		Get(gomock.AssignableToTypeOf(ctxType), plBefore.ID).
+		Return(plBefore, nil)
+	pluginMock.EXPECT().
+		NewDispenser(gomock.Any(), newDLQ.Plugin).
+		Return(pluginDispenser, nil)
+	pluginMock.EXPECT().
+		ValidateDestinationConfig(gomock.Any(), pluginDispenser, newDLQ.Settings).
+		Return(nil)
+	plsMock.EXPECT().
+		UpdateDLQ(gomock.AssignableToTypeOf(ctxType), plBefore.ID, newDLQ).
+		Return(want, nil)
+
+	got, err := orc.Pipelines.UpdateDLQ(ctx, plBefore.ID, newDLQ)
+	is.Equal(got, want)
+	is.NoErr(err)
+}
+
+func TestPipelineOrchestrator_UpdateDLQ_PipelineRunning(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+	db := &inmemory.DB{}
+	plsMock, consMock, procsMock, pluginMock := newMockServices(t)
+
+	plBefore := &pipeline.Instance{
+		ID:     uuid.NewString(),
+		Status: pipeline.StatusRunning,
+	}
+
+	orc := NewOrchestrator(db, log.Nop(), plsMock, consMock, procsMock, pluginMock)
+	plsMock.EXPECT().
+		Get(gomock.AssignableToTypeOf(ctxType), plBefore.ID).
+		Return(plBefore, nil)
+
+	got, err := orc.Pipelines.UpdateDLQ(ctx, plBefore.ID, pipeline.DLQ{})
+	is.Equal(got, nil)
+	is.True(err != nil)
+	is.Equal(pipeline.ErrPipelineRunning, err)
```

Comment:
The check `is.Equal(pipeline.ErrPipelineRunning, err)` also implies `is.True(err != nil)`, right?

Reply:
It does, it's a bit redundant 🤫